[ci skip] Fix docs for `default_form_builder`

Earlopain opened the pull request:
Passing a instance is not what you're supposed to do here. The instance will instead be passed to you in `form_with` and similar.

The top-level documentation already explains it correctly: https://api.rubyonrails.org/classes/ActionController/FormBuilder.html

p8 commented on actionmailer/lib/action_mailer/form_builder.rb:
```diff
@@ -23,7 +23,7 @@ module ClassMethods
       #
       # ==== Parameters
-      # * <tt>builder</tt> - Default form builder, an instance of ActionView::Helpers::FormBuilder
+      # * <tt>builder</tt> - Default form builder. Must be a subclass of ActionView::Helpers::FormBuilder
```
I don't think it needs to be a subclass, it just needs to have the same interface.
To keep inline with the documentation on line 7, maybe:
```suggestion
      # * <tt>builder</tt> - Default form builder. Accepts a subclass of ActionView::Helpers::FormBuilder
```

Earlopain replied:
Yes, that's better. Thanks

p8 commented on actionpack/lib/action_controller/form_builder.rb:
```diff
@@ -40,7 +40,7 @@ module ClassMethods
       #
       # #### Parameters
-      # *   `builder` - Default form builder, an instance of
+      # *   `builder` - Default form builder. Must be a subclass of
```
```suggestion
      # *   `builder` - Default form builder. Accepts a subclass of
```

Earlopain committed:
Fix docs for `default_form_builder`

Passing a instance is not what you're supposed to do here.
The instance will instead be passed to you in `form_with` and similar.

Co-authored-by: Petrik de Heus <petrik@deheus.net>

p8 commented:
Thanks @Earlopain !
